pkg/synchronizer/synchronizer.go

```diff
@@ -165,6 +164,38 @@ func (n *Synchronizer) Reconcile(ctx context.Context, req ctrl.Request, app reso
 		}
 	}()
 
+	if app.GetObjectMeta().GetDeletionTimestamp().IsZero() {
+		if !controllerutil.ContainsFinalizer(app, NaiseratorFinalizer) {
+			controllerutil.AddFinalizer(app, NaiseratorFinalizer)
+			err = n.Update(ctx, app)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if controllerutil.ContainsFinalizer(app, NaiseratorFinalizer) {
```

Comment:
Burde denne blokken vært en egen funksjon/metode for å samle logikken?

Reply:
Litt usikker på hva jeg skulle samlet den under. Prøvd å skille det som legger til og fjerner finalizer-en og hva vi faktisk gjør ved sletting av applikasjonen.

Kunne kanskje flyttet hele if-en (fra linje 167) ut en egen metode.

Reply:
Ta en titt på denne: https://github.com/nais/naiserator/pull/319/commits/d3f514bba3c9b7615f7b5e87e4508bbfd41f03d5

Reply:
Syns det ble veldig mye mer leselig :+1: